python/setup.py

```diff
@@ -250,14 +257,22 @@ def run(self):
         license='http://www.apache.org/licenses/LICENSE-2.0',
         # Don't forget to update python/docs/source/getting_started/install.rst
         # if you're updating the versions or dependencies.
-        install_requires=['py4j==0.10.9.2'],
+        install_requires=[
+            'py4j==0.10.9.2',
+            'pandas>=0.23.2',
+            'pyarrow>=0.10',
+            'numpy>=1.14,<1.20.0',
+        ],
         extras_require={
             'ml': ['numpy>=1.7'],
             'mllib': ['numpy>=1.7'],
             'sql': [
                 'pandas>=%s' % _minimum_pandas_version,
                 'pyarrow>=%s' % _minimum_pyarrow_version,
-            ]
+            ],
+            'pandas.mlflow': ['mlflow>=1.0'],
+            'pandas.plotly': ['plotly>=4.8'],
+            'pandas.matplotlib': ['matplotlib>=3.0.0,<3.3.0'],
```

Comment:
I'm not sure we should list these here? cc @HyukjinKwon

Reply:
I think we can remove these for now.

Reply:
Maybe we could:

```
extras_require={
    ...
    'pandas_on_spark':  [
        'pandas>=0.23.2',
        'pyarrow>=0.10',
        'numpy>=1.14,<1.20.0',
    ]
    ...
```

Yeah, I think we should remove them for now.

Reply:
We might need to add some description in [1] to add the dependencies info for pandas_on_spark.
[1] https://spark.apache.org/docs/latest/api/python/getting_started/install.html#dependencies

Reply:
Optional dependencies for pandas-on-spark have been removed.
Dependencies for pandas-on-spark have been moved to `extras_require`.

Reply:
Descriptions are added to `install.rst`.

Shall we call it `Pandas-on-spark` in the document?